benches/benches/src/sv1/criterion_sv1_benchmark.rs

```diff
@@ -0,0 +1,240 @@
+//! This code uses `criterion` crate to benchmark the performance sv1.
+//! It measures connection time, send subscription latency and share submission time.
+
+use criterion::{black_box, Criterion};
+use v1::{ClientStatus, IsClient};
+
+#[path = "./lib/client.rs"]
+mod client;
+use crate::client::{extranonce_from_hex, notify, *};
+
+fn benchmark_get_subscribe(c: &mut Criterion, mut client: Client) {
+    let mut group = c.benchmark_group("client-sv1-get-subscribe");
+    group.bench_function("client-sv1-get-subscribe", |b| {
+        client.status = ClientStatus::Configured;
+        let extranonce = Some(extranonce_from_hex("0000"));
+        b.iter(|| black_box(client.subscribe(black_box(10), extranonce.clone()).unwrap()));
+    });
+}
+
+fn benchmark_subscribe_serialize(c: &mut Criterion, mut client: Client) {
+    let mut group = c.benchmark_group("client-sv1-subscribe-serialize");
+
+    group.bench_function("client-sv1-subscribe-serialize", |b| {
+        client.status = ClientStatus::Configured;
+        let extranonce = Some(extranonce_from_hex("0000"));
+        let subscribe = client.subscribe(black_box(10), extranonce).unwrap();
+        b.iter(|| {
+            black_box(Client::serialize_message(black_box(subscribe.clone())));
+        });
+    });
+}
+
+fn benchmark_subscribe_serialize_deserialize(c: &mut Criterion, mut client: Client) {
+    let mut group = c.benchmark_group("client-sv1-subscribe-serialize-deserialize");
+
+    group.bench_function("client-sv1-subscribe-serialize-deserialize", |b| {
+        client.status = ClientStatus::Configured;
+        let subscribe = client.subscribe(black_box(10), None).unwrap();
+        let serialized = Client::serialize_message(black_box(subscribe));
+        b.iter(|| {
+            black_box(Client::parse_message(black_box(serialized.clone())));
+        });
```

Comment:
From your [README](https://github.com/stratum-mining/stratum/blob/b6b997ac52b55448f8dc5fe960f4176ccfb6c252/benches/benches/README.md) : `"client_sv1_serialize_deserialize: Measures the latency and system requirements it takes to serialize and then deserialize a subscription message."`

Actually here it seems to me that what is really benchmarked is the deserialization(parsing) of an already serialized subscription message. Since the iter() function contains only the parse_message function, the impact of the subscription and serialization of it is not considered in the benchmark, imo.
Please @Fi3 if you can have a look here to confirm/deny.

Reply:
That would be better, it means that we do not have to subtract the time for subscribe serialize and deserialize when we show the results. 

From the code I would say that you are right. If from the documentation is not clear if what is outside `iter` is benchmarked @Priceless-P to be 100% sure just try to "mess" (maybe add a loop or something similar) with the part that is outside `iter` and check if the result change

Reply:
@GitGab19  The impact of the subscription and serialization is considered in 
[`benchmark_get_subscribe`](https://github.com/Priceless-P/stratum/blob/523a9c795fe4750a1c35619f0922d8ebab19b592/benches/benches/src/sv1/criterion_sv1_benchmark.rs#L11) and [`benchmark_subscribe_serialize`](https://github.com/Priceless-P/stratum/blob/523a9c795fe4750a1c35619f0922d8ebab19b592/benches/benches/src/sv1/criterion_sv1_benchmark.rs#L20) respectively
 
 
 Also, @Fi3  I experimented by adding a loop before `iter` function,  like so
 ```rust
 fn benchmark_subscribe_serialize_deserialize(c: &mut Criterion, mut client: Client) {
    let mut group = c.benchmark_group("client-sv1-subscribe-serialize-deserialize");
    group.bench_function("client-sv1-subscribe-serialize-deserialize", |b| {
        let mut _count = 0;
        for _ in 0..100 {
            _count += 1;
        }
        client.status = ClientStatus::Configured;
        let subscribe = client.subscribe(black_box(10), None).unwrap();
        let serialized = Client::serialize_message(black_box(subscribe));
        b.iter(|| {
            black_box(Client::parse_message(black_box(serialized.clone())));
        });
    });
}
```
This did not change the result of the benchmark, indicating that only what is included in the `iter` function is benchmarked.